### PR TITLE
 Fixes #379

### LINKED ISSUE #379: Nudging breaks sync messages with an execution
This issue happens when #371 is merged. Still, I wanted to open a new issue on that as I think there is more involved here.

Given the following scenario:
image

When we now add a new sync message after the existing reply message from `Lifeline2` to `Lifeline1` the following happens:
image

### PR DESCRIPTION
**Pull request by planger**
This branch builds on the branch #371 and aims to address #379. Unfortunately, this fix breaks SelfMessageReconnectionUITest and I'm not sure whether I understand well enough the functionality this test aims to verify.

@cdamus Can you please have a look? If you see what's wrong, please feel free to fix the entire issue. Otherwise, please let me know what the functionality is exactly that this change breaks. Thanks a lot!

### CODE REVIEW

**Comment by cdamus** [comment]
The reason why the tests fail is that the creation of the self-message results in an invalid semantic ordering of fragments.  In the diagram under test, starting with this:

screen shot 2018-10-03 at 16 22 19

when the self-message is created, this is the result visually:

screen shot 2018-10-03 at 16 22 36

with this ordering in the semantics that doesn't correspond to the time ordering in the visuals:

screen shot 2018-10-03 at 16 59 29

The new message occurrences should be ahead of the starting execution occurrence, according to the visuals.

**Comment by planger** [comment]
Thanks a lot again @cdamus for fixing the bug that broke the test!

I encountered a few other problems while testing and think that they are caused by a wrong y-position reference that we were using for computing the direct "graphical" predecessor and successor element. We were using the one that may have already shifted down to ensure minimum padding. This may cause the actual y-reference to be shifted beyond the y-position of the next element in certain scenarios. E.g. if you create an execution on the top of a lifeline and try to create a sync message (with execution to be created) from another lifeline right above it:
image

This led to the following result:
image

I pushed a refactoring to accommodate a fix (b3b2607) and a proposal for a fix (f85d245).

File: plugins/org.eclipse.papyrus.uml.interaction.model/src/org/eclipse/papyrus/uml/interaction/internal/model/commands/InsertMessageCommand.java [hunk 1]
```diff
@@ -812,7 +813,10 @@ private void validateOffsetChange(Optional<MExecution> exec, int absoluteY, int
 	}
 
 	protected Predicate<? super MElement<?>> isGraphicallyBefore(OptionalInt yPosition) {
-		return e -> e.getBottom().orElse(Integer.MAX_VALUE) < yPosition.orElse(Integer.MAX_VALUE);
+		return e -> {
+			System.out.println("existing at " + e.getBottom() + " ref: " + yPosition);
```
Comment (cdamus) [inline plugins/org.eclipse.papyrus.uml.interaction.model/src/org/eclipse/papyrus/uml/interaction/internal/model/commands/InsertMessageCommand.java]: Just remove this, and we're good!

**Comment by planger** [comment]
@cdamus Thanks a lot for the fast feedback and for catching the sys-out. ce80061 removes this left-over from debugging. Please approve if you're fine with this. Thanks!

**Pull request merged**